Core/dp 20570 design tokens

tkoleary opened the pull request:
## Description
Updated the  colors page and the logo page layout and content plus some minor styling

Logo:
- Simplified language
- Altered layout to be more compact by moving download links into canvas
- Added background color specification per logo
- Added min and max to logo sizes
- Changed heading on spacing to 'Clear space' 

Color:
- Changed nomenclature 'gradients' to tints and shades to remove ambiguity with css gradients
- Changed nomenclature from 'Utility' to 'Semantic' colors
- Simplified Primary, Alt, highlight into a more compact 'color hierarchy' section

## Related Issue / Ticket
- [JIRA issue](DP-20570)

## Steps to Test
1. Switch the branch
2. Run `rush start:core`
3. Go to docs/design-tokens-colors and compare to screenshots
4. Go to docs/design-tokens-logo and compare to screenshots

## Screenshots
<img width="1110" alt="Screen Shot 2020-12-02 at 3 45 20 PM" src="https://user-images.githubusercontent.com/3894652/100935645-f9d2d180-34b5-11eb-8236-94eebc16267c.png">
<img width="1045" alt="Screen Shot 2020-12-02 at 3 45 31 PM" src="https://user-images.githubusercontent.com/3894652/100935638-f93a3b00-34b5-11eb-9f59-55aea4f0287a.png">
<img width="1056" alt="Screen Shot 2020-12-02 at 3 45 54 PM" src="https://user-images.githubusercontent.com/3894652/100935632-f63f4a80-34b5-11eb-8618-35710d32187f.png">

<img width="1057" alt="Screen Shot 2020-12-02 at 3 46 09 PM" src="https://user-images.githubusercontent.com/3894652/100935999-7d8cbe00-34b6-11eb-8d93-ca7581ff0faf.png">
<img width="1058" alt="Screen Shot 2020-12-02 at 3 46 22 PM" src="https://user-images.githubusercontent.com/3894652/100935994-7c5b9100-34b6-11eb-910f-2f2a2a8f93c6.png">
<img width="1046" alt="Screen Shot 2020-12-02 at 3 46 33 PM" src="https://user-images.githubusercontent.com/3894652/100935989-7a91cd80-34b6-11eb-9010-01e0393c59e0.png">
<img width="1043" alt="Screen Shot 2020-12-02 at 3 46 44 PM" src="https://user-images.githubusercontent.com/3894652/100935981-78c80a00-34b6-11eb-8766-b5b7fee8cecd.png">




#### @TODO
Could usee some style follow-up particularly margin in the canvas

tkoleary committed:
Updates to the colors story

tkoleary committed:
Updates to logo page content and layout

clairesunstudio commented on packages/core/stories/tokens/logo/_logo-display.scss:
```diff
@@ -0,0 +1,8 @@
+@use "00-base/configure" as *;
+@use "sass:color" as *;
```
I don't think we need this file. Storybook doesn't encapsulate styles even though you imported into different stories separately. Hence we have a index.scss, and this class already exist in color-display, as long as it's imported into a story file you don't need to import it again for a different story otherwise it will be duplicated in production.

clairesunstudio replied:
storybook doesn't encapsulate styles well, this will override the .sg-swatch class in colors

clairesunstudio committed:
Merge branch 'core/storybook-integration' into core/DP-20570-design-tokens

clairesunstudio commented on packages/core/stories/tokens/colors/Colors.stories.mdx:
```diff
@@ -1,98 +1,72 @@
   }
 </ul>
 
-## Theme Colors
```
I think we should name it "theme colors" to keep it consistent with the prop name for components, e.g. https://mayflower.digital.mass.gov/react/index.html?path=/story/atoms-buttons-button--button-example

clairesunstudio commented on packages/core/stories/tokens/colors/colors.json:
```diff
@@ -78,54 +73,28 @@
     }
   ],
-  "primaryColors": [
+	"hierarchicalColors": [
```
As we discussed earlier, we need a place to document the lighter color variants that are used in the system 
https://mayflower.digital.mass.gov/patternlab/?p=base-colors

clairesunstudio commented on packages/core/stories/tokens/colors/Colors.stories.mdx:
```diff
@@ -1,98 +1,72 @@
 
-**Examples:** Headings, titles, icons, graphics, calls to action, buttons,
-and backgrounds.
+Some brand colors like `$c-duckling-ellow` and `$c-berkshires-green` are doing double duty as Semantic colors while others like `$c-bay-blue` and `$c-independence-cranberry` are not. We wanted to keep our palett compact, but be careful not to use `$c-bay-blue` as a focus color or `$c-independence-cranberry` as a warning color.
```
I think it's easier to talk through all the changes. I like what you added here. But let's keep the overall structure for now
```suggestion
Some brand colors like `$c-duckling-ellow` and `$c-berkshires-green` are doing double duty as Semantic colors while others like `$c-bay-blue` and `$c-independence-cranberry` are not. We wanted to keep our palette compact, but be careful not to use `$c-bay-blue` as a focus color or `$c-independence-cranberry` as a warning color.
```

tkoleary replied:
fixed on my branch

tkoleary committed:
Updates to pr per Minghua

-reverted hierarchical to primary, alt, highlight sections
-moved down semantic colors to below theme colors
-fixed typo 'palette'

tkoleary committed:
Removed logo display file

clairesunstudio committed:
pull

clairesunstudio commented on packages/core/stories/tokens/colors/Colors.stories.mdx:
```diff
@@ -1,36 +1,32 @@
-brand on all public-facing assets and documents.
-Organizations that have their own visual identity are invited to extend this color palette with additional colors that reflect their unique brand in conjunction with the shared Commonwealth identity.
+
+## Theme Colors
```
```suggestion
## Brand Colors
```

clairesunstudio commented on packages/core/stories/tokens/colors/Colors.stories.mdx:
```diff
@@ -1,36 +1,32 @@
+### Color hierarchy
 
-### Primary Colors
+`$c-bay-blue` is the **Primary Color** is the default color for links and primary action buttons. Accompanying shades and tints can be used to create depth. It's also used in headings, titles, icons, graphics, and calls to action. `$c-berkshires-green` is the **Alternate Primary Color**, it usually does the same job as the primary color, but when the surrounding context is different, for example  ...
```
```suggestion
`$c-bay-blue`, as the **Primary Color**, is the default color for links and primary action buttons. Accompanying shades and tints can be used to create depth. It's also used in headings, titles, icons, graphics, and calls to action. `$c-berkshires-green` is the **Alternate Primary Color**, it usually does the same job as the primary color, but when the surrounding context is different, for example in a call-to-action block. `$c-duckling-ellow` is the **Highlight Color** and it's used for time sensitive information like updates and alerts.
```

clairesunstudio committed:
tweaks and fixes

clairesunstudio committed:
tweaks and fixes

clairesunstudio committed:
capitalization

clairesunstudio committed:
match variable names

clairesunstudio commented on packages/core/stories/tokens/logo/Logo.stories.mdx:
```diff
@@ -31,115 +31,98 @@ According the [public records on the Massachusetts state seal](https://www.sec.s
+| Medium | Recommended Size | Minimum size | Maximum size |
+|---|---|---|---|
+| **Print (in)** | 0.5 x 0.5 | 1 x 1 | 2 x 2 |
+| **Digital (px)** | 32 x 32 | 45 x 45 | 200 x 200 |
```
the recommended sizes are smaller than the minimum sizes?

tkoleary replied:
oops. No that's wrong.

tkoleary replied:
Fixed

clairesunstudio committed:
tweaks and fixes for logo docs

tkoleary committed:
swapped incorrectly oredered size labels

tkoleary committed:
Merge branch 'core/DP-20570-design-tokens' of https://github.com/massgov/mayflower into core/DP-20570-design-tokens

clairesunstudio committed:
add additional background color usage info

clairesunstudio committed:
revert label change

clairesunstudio reviewed (approved):
approved!
